lib/kernel/doc/src/net_kernel.xml

```diff
@@ -345,28 +349,114 @@ $ <input>erl -sname foobar</input></pre>
     </func>
 
     <func>
-      <name since="">start([Name]) -> {ok, pid()} | {error, Reason}</name>
-      <name since="">start([Name, NameType]) -> {ok, pid()} | {error, Reason}</name>
-      <name since="">start([Name, NameType, Ticktime]) -> {ok, pid()} | {error, Reason}</name>
+      <name name="start" arity="2" since="OTP @OTP-17905@"/>
       <fsummary>Turn an Erlang runtime system into a distributed node.</fsummary>
-      <type>
-        <v>Name = atom()</v>
-        <v>NameType = shortnames | longnames</v>
-        <v>Reason = {already_started, pid()} | term()</v>
-      </type>
       <desc>
-        <p>Turns a non-distributed node into a distributed node by
-          starting <c>net_kernel</c> and other necessary processes.</p>
-        <p>Notice that the argument is a list with exactly one, two, or
-          three arguments. <c>NameType</c> defaults to <c>longnames</c>
-          and <c>Ticktime</c> to <c>15000</c>.
+        <p>
+          Turns a non-distributed node into a distributed node by
+          starting <c>net_kernel</c> and other necessary processes.
         </p>
+
+        <p>
+          <c>Name</c> determines the part of the node name preceding the
+          <c>@</c> sign. If <c>Name</c> equals to <em><c>undefined</c></em>,
+          the distribution will be started to request a dynamic node name from the
+          first node it connects to. See
+          <seeguide marker="system/reference_manual:distributed#dyn_node_name">
+            Dynamic Node Name</seeguide>. The only characters allowed in <c>Name</c>
+            are:
+        </p>
+        <list>
+          <item><c>a-z</c></item>
+          <item><c>A-Z</c></item>
+          <item><c>0-9</c></item>
+          <item><c>-</c></item>
+          <item><c>_</c></item>
+        </list>
```

Comment:
I propose the following alternative text. 
The allowed characters in <c>Name</c> are <c>[a-z] | [A-Z] | [0-9] '-' | '_'</c>.
Which when rendered could look something like: 
The allowed characters in `Name` are `[a-z] | [A-Z] | [0-9] '-' | '_'`.